python/tank/bootstrap/configuration.py

```diff
@@ -92,10 +92,20 @@ def get_tk_instance(self, sg_user):
         # by creating a pipeline configuration object directly
         # and pass that into the factory method.
 
-        pc = pipelineconfig.PipelineConfiguration(path, self.descriptor)
+        if self._does_pipelineconfigution_supports_descriptor(pipelineconfig.PipelineConfiguration):
+            pc = pipelineconfig.PipelineConfiguration(path, self.descriptor)
```

Comment:
While having a setter would be easier to detect (via hasattr), what worries me is that when we create a pipeline configuration instance in the future we also need to set an extra thing after, which is error prone. What's the point of an init method if it doesn't completely init the object.

With that being said, I thing there is a much cleaner approach actually! I'll update the PR right away :)